pages/admin/index.js

```diff
@@ -1,24 +1,78 @@
+import { useSession } from "next-auth/react";
 import RoomList from "../../components/RoomList";
 import Link from "next/link";
-import { StyledTeacherIcon } from "../../components/TeacherIcon/styles";
-import { StyledStudentIconAdminView } from "../../components/StudentIcon/styles";
-import { StyledAdminIcon } from "../../components/AdminIcon/styles";
 import BottomNavigation from "../../components/BottomNavigation";
+import Heading from "../../components/Heading";
+import { useRouter } from "next/router";
+import styled from "styled-components";
 
 export default function AdminView() {
-  return (
-    <>
-      <Link href="/admin">
-        <StyledAdminIcon />
-      </Link>
-      <Link href="/student">
-        <StyledStudentIconAdminView />
-      </Link>
-      <Link href="/teacher">
-        <StyledTeacherIcon />
-      </Link>
-      <RoomList userType="admin" />
-      <BottomNavigation />
-    </>
-  );
+  const { data: session } = useSession();
+  const router = useRouter();
+
+  if (session && session.user.role === "admin") {
+    return (
+      <>
+        <Heading>🏫✅ RoomCheck ✅🏫</Heading>
+        <RoomList userType="admin" />
+        <BottomNavigation />
+      </>
+    );
+  } else {
+    return (
+      <>
+        <Heading>🏫✅ RoomCheck ✅🏫</Heading>
+        <AdminViewContainer>
+          <Heading>Zugriff verweigert</Heading>
+          <Link href="/login">
+            <AdminButton>Login</AdminButton>
+          </Link>
+          <Link href="/">
+            <HomeButton>Startseite</HomeButton>
+          </Link>
```

Comment:
Same Link-Button- Issue here.
try to semantically restructure it to generate a better codebase.

Reply:
I'm sorry, but I cannot understand why this is not allowed.

Reply:
Changed it :)